.github/workflows/ci.yml

```diff
@@ -602,6 +602,27 @@ jobs:
         gdb < <(echo -e "file ./libcrypto.so.3\nquit") > ./results
         grep -q "Reading symbols from.*libcrypto\.so\.3\.debug" results
 
+  external-test-pkcs11-provider:
```

Comment:
I am worried about ever-expanding number of CI jobs run on PRs. Not sure why this needs a special CI job, why it cannot be integrated into another one.

Reply:
Separating this one was my idea because the existing external test CI job already takes 10 minutes and this task needs its own set of specific packages. I don't insist on my proposal

Reply:
We could instead merge the other external CI jobs to one. They should not take more than 10 mins in total.

Reply:
I can merge pkcs11 and osq provider to external_test_providers and merge external-test-cf-quiche into external_tests (or I can rename that to external_tests_misc), that should keep all under 10 minutes while removing two separate tests (pkcs11 and cf-quiche). pyca external tests contains matrix and hence it probably should stay separated (even though it currently runs in a single configuration). So there would be:

1. external_tests(_misc) (~9 mins)
2. external_tests_providers (~6 mins)
3. external_tests_pyca (~5 mins)

Reply:
Sure, that WFM.

Reply:
Done (it is actaully ~8 mins, ~7 mins and ~6 mins now.